paddle/framework/tensor_impl.h

```diff
@@ -117,6 +117,8 @@ inline void Tensor::CopyFrom(const Tensor& src,
     memory::Copy(boost::get<platform::GPUPlace>(dst_place), dst_ptr,
                  boost::get<platform::GPUPlace>(src_place), src_ptr, size, 0);
   }
+  PADDLE_ENFORCE(cudaStreamSynchronize(0),
+                 "cudaStreamSynchronize failed in Tensor CopyFrom");
```

Comment:
It's better to move `cudaStreamSynchronize` to pybind. And it's not good to use 0 as cuda stream in the `memory::Copy` interface.

Reply:
Yes, I will fix it later. And I set up a issue #3796 , try to add a DeviceContextManager to manage CUDA streams appropriately.